core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -0,0 +1,220 @@
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Function;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+import com.linecorp.armeria.common.HttpHeaders;
+
+/**
+ * Builds a new logging decorator.
+ */
+public class LoggingDecoratorBuilder<T extends LoggingDecoratorBuilder<T>> {
+    private static final Function<HttpHeaders, HttpHeaders> DEFAULT_HEADERS_SANITIZER = Function.identity();
+    private static final Function<Object, Object> DEFAULT_CONTENT_SANITIZER = Function.identity();
+
+    private LogLevel requestLogLevel = LogLevel.TRACE;
+    private LogLevel successfulResponseLogLevel = LogLevel.TRACE;
+    private LogLevel failedResponseLogLevel = LogLevel.WARN;
+    private Function<HttpHeaders, HttpHeaders> requestHeadersSanitizer = DEFAULT_HEADERS_SANITIZER;
+    private Function<Object, Object> requestContentSanitizer = DEFAULT_CONTENT_SANITIZER;
+    private Function<HttpHeaders, HttpHeaders> responseHeadersSanitizer = DEFAULT_HEADERS_SANITIZER;
+    private Function<Object, Object> responseContentSanitizer = DEFAULT_CONTENT_SANITIZER;
+    private float samplingRate = 1.0f;
+
+    /**
+     * Sets the {@link LogLevel} to use when logging requests. If unset, will use {@link LogLevel#TRACE}.
+     */
+    public T requestLogLevel(LogLevel requestLogLevel) {
+        this.requestLogLevel = requireNonNull(requestLogLevel, "requestLogLevel");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link LogLevel} to use when logging requests.
+     */
+    public LogLevel getRequestLogLevel() {
+        return requestLogLevel;
+    }
+
+    /**
+     * Sets the {@link LogLevel} to use when logging successful responses (e.g., no unhandled exception).
+     * If unset, will use {@link LogLevel#TRACE}.
+     */
+    public T successfulResponseLogLevel(LogLevel successfulResponseLogLevel) {
+        this.successfulResponseLogLevel =
+                requireNonNull(successfulResponseLogLevel, "successfulResponseLogLevel");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link LogLevel} to use when logging successful responses (e.g., no unhandled exception).
+     */
+    public LogLevel getSuccessfulResponseLogLevel() {
+        return successfulResponseLogLevel;
+    }
+
+    /**
+     * Sets the {@link LogLevel} to use when logging failure responses (e.g., failed with an exception).
+     * If unset, will use {@link LogLevel#WARN}.
+     */
+    public T failureResponseLogLevel(LogLevel failedResponseLogLevel) {
+        this.failedResponseLogLevel = requireNonNull(failedResponseLogLevel, "failedResponseLogLevel");
+        return unsafeCast(this);
+    }
+
+    /**
+     * Returns the {@link LogLevel} to use when logging failure responses (e.g., failed with an exception).
+     */
+    public LogLevel getFailedResponseLogLevel() {
```

Comment:
- Remove `get`?
- protected? (a subclass can override it to make it accessible from the classes from the same package)